lib/fiddle/struct.rb

```diff
@@ -240,10 +310,11 @@ def [](*args)
       when Array
         case ty[0]
         when TYPE_VOIDP
-          val = val.collect{|v| Pointer.new(v)}
+          val = val.collect{|v| v = Pointer.new(v); v.size = SIZEOF_VOIDP; v }
```

Comment:
`Pointer#size` is not the size of address. It's the size of area that is pointed by the address.